push/sender/src/test/java/org/jboss/aerogear/unifiedpush/message/jms/TestTriggerMetricCollectionDeduplication.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.jms;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Resource;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.jms.Queue;
+
+import org.jboss.aerogear.unifiedpush.message.AbstractJMSTest;
+import org.jboss.aerogear.unifiedpush.message.event.TriggerMetricCollectionEvent;
+import org.jboss.aerogear.unifiedpush.test.archive.UnifiedPushArchive;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class TestTriggerMetricCollectionDeduplication extends AbstractJMSTest {
+
+    @Deployment
+    public static WebArchive archive() {
+        return UnifiedPushArchive.forTestClass(TestTriggerMetricCollectionDeduplication.class)
+                .withMessaging()
+                .withMessageDrivenBeans()
+                .addClasses(CdiJmsBridge.class)
+                .as(WebArchive.class);
+    }
+
+    @Inject @DispatchToQueue
+    private Event<TriggerMetricCollectionEvent> triggerMetricCollection;
+
+    @Resource(mappedName = "java:/queue/TriggerMetricCollectionQueue")
+    private Queue triggerMetricCollectionQueue;
+
+    private static final ConcurrentLinkedQueue<TriggerMetricCollectionEvent> receivedTriggers = new ConcurrentLinkedQueue<TriggerMetricCollectionEvent>();
+    private static final CountDownLatch firstMessageLatch = new CountDownLatch(1);
+    private static String messageId;
+
+    @Test(timeout = 5000)
+    public void testDeduplication() throws InterruptedException {
```

Comment:
What kind of duplication is it testing? Is it asserting that metrics _are_ or _aren't_ duplicated?